Source/DafnyCore/Options/CommonOptionBag.cs

```diff
@@ -142,6 +142,12 @@ public class CommonOptionBag {
 true - The char type represents any Unicode scalar value.".TrimStart()) {
   };
 
+  public static readonly Option<bool> TypeSystemRefresh = new("--type-system-refresh", () => false,
+    @"
+false - The type-inference engine and supported types are those of Dafny 4.0.
+true - Use an updated type-inference engine. Warning: This mode is under construction and probably won't work at this time.".TrimStart()) {
+  };
```

Comment:
```suggestion
    IsHidden = true
  };
```

Given that this option is very much for our own use (and even the name will become dated at some point), making it hidden seems appropriate.

Reply:
I don't know if there's a way to do a similar thing for the legacy version of the option.